src/math/fibonacci.lua

```diff
@@ -0,0 +1,80 @@
+-- Fibonacci.lua
+
+-- Fibonacci Sequence:
+-- The Fibonacci sequence is a series of numbers where each number (after the first two) is the sum of the two preceding ones.
+-- More information: https://en.wikipedia.org/wiki/Fibonacci_number
+
+-- Author: Gyandeep
+
+--      Implemented nth Fibonacci number using different approaches in O(2^N) , O(N) , O(logN) , O(1) .
+
+
+-- Function to calculate Fibonacci numbers using recursion
+-- Time Complexity: O(2^n)
```

Comment:
Inaccurate. Exponential, but not 2^n (it grows with the golden ratio). I'd just use `--! exponential time complexity, just for illustrative purposes` here.